chiselFrontend/src/main/scala/chisel3/core/ChiselAnnotation.scala

```diff
@@ -28,3 +28,40 @@ case class ChiselAnnotation(component: InstanceId, transformClass: Class[_ <: Tr
     }
   }
 }
+
+/** Marks that a signal should not be removed by Chisel and Firrtl optimization passes
+  *
+  * @example {{{
+  * class MyModule extends Module {
+  *   val io = IO(new Bundle {
+  *     val a = Input(UInt(32.W))
+  *     val b = Output(UInt(32.W))
+  *   })
+  *   io.b := io.a
+  *   val dead = io.a +% 1.U // normally dead would be pruned by DCE
+  *   dontTouch(dead) // Marking it as such will preserve it
```

Comment:
As far as I know, we can't do a `@something`-style macro annotation inline, those can be applied to only types and classes.
You could make `dontTouch` a macro transform, though you don't gain anything in this case.